source/adios2/core/IO.tcc

```diff
@@ -260,6 +238,41 @@ ADIOS2_FOREACH_STDTYPE_2ARGS(make_GetVariableMap)
 ADIOS2_FOREACH_ATTRIBUTE_STDTYPE_2ARGS(make_GetAttributeMap)
 #undef make_GetAttributeMap
 
+template <class T>
+Params IO::GetVariableInfo(const std::string &variableName,
+                           const std::set<std::string> &keys)
+{
+    Params info;
+    const std::set<std::string> keysLC = helper::LowerCase(keys);
+    // small strings so passing by value
+    auto lf_Insert = [&](const std::string key, const std::string value) {
+        const std::string keyLC = helper::LowerCase(key);
+        if (keys.empty() || keysLC.count(keyLC) == 1)
+        {
+            info[key] = value;
+        }
+    };
+
+    Variable<T> &variable = *InquireVariable<T>(variableName);
+
+    lf_Insert("Type", variable.m_Type);
+    lf_Insert("AvailableStepsCount",
+              helper::ValueToString(variable.m_AvailableStepsCount));
+    lf_Insert("Shape", helper::VectorToCSV(variable.Shape()));
```

Comment:
variable.Shape() is the call in the original GetAvailableVariables() function that called the engine.BlockInfo<T>() for every variable which went through all the blocks. I don't see how this new function avoids calling that all the time.

Reply:
@pnorbert there is a conditional inside `lf_Insert`. Each field is populated only when the key exists in keys or if the keys set is empty (current default). The added test in this PR verifies this.

Reply:
I could add a key for `name` to only return the map with name keys and empty sub-maps values without breaking the API. What do you think?

Reply:
lf_Insert is too late. You are passing the result of the variable.Shape() as string, so you already parsed the blocks. I think instead of the lambda, an explicit switch case should be written in the main body of the function and only call variable.Shape() if there is a key for it.

Reply:
I see, good catch. My bad, I missed the input l'll move the conditional outside.